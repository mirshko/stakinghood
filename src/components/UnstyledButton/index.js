import React from "react";
import styled from "styled-components";

const Button = styled.button`
  margin: 0;
  padding: 0;
  border: none;
  font: inherit;
  color: inherit;
  appearance: none;
  background: none;
  display: inline-block;
  outline: none;
  text-align: inherit;
  cursor: pointer;

  width: ${props => (props.width ? props.width : "100%")};

  & > * {
    transition: opacity 0.125s ease;
  }

  &:hover > * {
    opacity: 0.64;
  }
`;

const UnstyledButton = ({ onClick, children, width }) => (
  <Button width={width} onClick={onClick}>
    {children}
  </Button>
);

export default UnstyledButton;
