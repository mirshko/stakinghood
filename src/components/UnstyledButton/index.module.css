.button {
  all: unset;
  display: block;
  cursor: pointer;
  width: 100%;
}

.button:hover > * {
  opacity: 0.64;
}

.button > * {
  transition: opacity 0.125s ease;
}
