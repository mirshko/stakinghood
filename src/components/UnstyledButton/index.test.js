import React from "react";
import { shallow } from "enzyme";

import UnstyledButton from "./index";

describe("<UnstyledButton />", () => {
  it("it works", () => {
    const wrapper = shallow(
      <UnstyledButton onClick={() => console.log("Clicked")}>
        Button
      </UnstyledButton>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toContain("Button");
  });
});
