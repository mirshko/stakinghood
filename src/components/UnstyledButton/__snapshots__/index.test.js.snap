// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnstyledButton /> it works 1`] = `
<styled.button
  onClick={[Function]}
>
  Button
</styled.button>
`;
